src/Lokad.Tokenizers/Tokenizer/BaseTokenizer.cs

```diff
@@ -835,4 +1019,33 @@ private List<string> DecodeToVec(List<long> tokenIds, bool skipSpecialTokens)
         }
         return tokens;
     }
+}
+
+public interface IMultiThreadedTokenizer<T> where T : IVocab
```

Comment:
Do we even need the multi-threaded version? I would rather be of the opinion of putting this aside entirely. The mono-threaded performance (once done correctly) should be sufficient.